### DISCUSSION TITLE
Welcome to AlphaFamImpute Discussions!

### DISCUSSION THREAD

**Discussion by gregorgorjanc** [AlphaGenes/AlphaFamImpute#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rjsdotorg** [comment]
Hello!
I've been programming for 40 years or so, and Python for 20, but recently got interested in genetic genealogy programming with a Visual Phasing[ set of tools](https://www.facebook.com/groups/1948458178832407) where I occasionally help the main author.

For my own personal 3-sibling Visual Phasing project, at this point out of my ~246 recombination points only ~11 have a problematic or ambiguous grandparent assignment so I'm pretty happy with it.
It seems to me that since these are based upon our 3 sibling build-37 files that it should now be possible to chop up the sibling kit files (consumer MyHeritage kits in this case) and reassemble into parent and grandparent kits, using the information now in the Excel workbook.
However, the rules for doing that are not entirely clear to me yet. Given
> rsid chromosome position allele1 allele2
> rs547237130 1 72526 A A
> ...

by what algorithm do we use the table values:
Chromosome, Start Mb, Finish Mb, No. SNPs, Length (cM)
and create the parental assignments? (Some might need modification since the text tables do not always match with the columns I now have due to false RPs or hidden RPs. I can deal with that.)
In the case of parents/grandparents of course there are many segments where one or two of the 4 grandparents do not appear, but, still are often available from 1st cousins' DNA.

Any direction is appreciated.

**Reply by gregorgorjanc** [reply]
@rjsdotorg I don’t fully follow your question, but note that this algorithm and software was designed for large sib families, so the accuracy of phasing and imputation of the parents will not be high with just 3 (or is it 4?) siblings - see https://academic.oup.com/view-large/figure/409074051/BIOINFORMATICS_36_15_4369_f1.png and more generally the full paper https://academic.oup.com/bioinformatics/article/36/15/4369/5848409

**Comment by rjsdotorg** [comment]
I'd like to try for two reasons:

- compare to the visual phasing result
- understand the concepts to possibly help with creating a parental DNA reconstruction application which utilizes the visual phasing result

I'm reading the [AlphaFamImpute](https://github.com/AlphaGenes/AlphaFamImpute/tree/ba0c70c367aa055fba7b4457986a5c42d409d110)/[docs](https://github.com/AlphaGenes/AlphaFamImpute/tree/ba0c70c367aa055fba7b4457986a5c42d409d110/docs)/[source](https://github.com/AlphaGenes/AlphaFamImpute/tree/ba0c70c367aa055fba7b4457986a5c42d409d110/docs/source) /index.rst
The pedigree.txt would be simply:
```
id1 0 0
id2 0 0
id3 0 0
id4 0 0
```
For the other files I'll need to write a short script to translate the consumer kit files to the map.txt, sequence.txt and genotype.txt
map.txt would be `chromosome position' with [rsid mapped to an int](https://www.biostars.org/p/253084/) from GRCh37 from the NCBI dbdirect from the kit file.

Would sequence.txt be similar [to this](https://support.bioconductor.org/p/9144015/), for each position?
(BTW I think this is a typo: "_The sequence data file is in a similar Sequence data is given in a similar format to the genotype data._")

genotype.txt I am not clear on.



Also, others [have proposed the reconstruction idea](https://thegeneticgenealogist.com/2016/11/27/visual-phasing-an-example-part-5-of-5/#:~:text=They%20share%20a%20much%20larger,the%20information%20from%20visual%20phasing!&text=I%20envision%20two%20major%20developments,rely%20only%20on%20matching%20segments.) but none have implemented it AFAIK:

> **1. Visual Phasing Automation.**  Visual Phasing will be partially or entirely automated. The data from three siblings could be fed into an algorithm that determines recombination points and fills in the chromosomes. Then, you could feed known matches into the algorithm to have it automatically identify which segments came from which grandparent.
> 
> **2. Grandparent Reconstruction.** The data from Visual Phasing, particularly from people lucky enough to have multiple siblings (4 or more) tested, could be used to recreate incredibly complete and accurate reconstructions of our grandparents. For example, we could extract the Visually Phased segments from one or more of the grandparents to create a grandparent kit. And since we’re using entire portions of chromosomes beyond matching segments, this will allow for greater reconstruction than tools like Lazarus which rely only on matching segments.

**Comment by rjsdotorg** [comment]
I have also contacted the authors of [HAPI-RECAP](https://pmc.ncbi.nlm.nih.gov/articles/PMC11118276) and joined their mailing list but have not heard from them. It seems more directly related to what I am after. [Their](https://hapi-dna.org/lab/) previous phasing work HAPI2 https://hapi-dna.org/hapi is online, but requires one parent.

**Comment by rjsdotorg** [comment]
[gregorgorjanc](https://github.com/gregorgorjanc) - a few questions:
I can generate the **sequence.txt** from my local copy of Homo_sapiens.GRCh37.dna.chromosome.1.fa for the appropriate loci. There are 609,347 loci.

**map.txt** would be `chromosome position' with [rsid mapped to an int](https://www.biostars.org/p/253084/) from GRCh37 from the NCBI dbdirect from the kit file.

**pedigree.txt** apparently will be zeros, as there is no input data for parents (or do parents still need an id?):
```
id1 0 0
id2 0 0
id3 0 0
id4 0 0
```

Re the **genotype,txt**, is it actually 0, 1, or 2 (or 9 if missing) or [A, C, G, T, 9]?

**Reply by gregorgorjanc** [reply]
> **pedigree.txt** apparently will be zeros, as there is no input data for parents (or do parents still need an id?):
> 
> ```
> id1 0 0
> id2 0 0
> id3 0 0
> id4 0 0
> ```

Since you want parent reconstruction, you should add parents too. Assuming the four are full sibs, you would do:

```
A 0 0
B 0 0 
id1 A B
id2 A B
id3 A B
id4 A B
```
 
> Re the **genotype,txt**, is it actually 0, 1, or 2 (or 9 if missing) or [A, C, G, T, 9]?

[0, 1, 2, or 9=missing].

We recently polished docs for AlphaPeel at https://alphapeel.readthedocs.io/en/latest/usage.html#file-formats - this is not AlphaFamImpute, but much of the code base is shared. This is work in progress, but I think the docs will be useful - let us know if you hit issues.
